sectionproperties/pre/sections.py

```diff
@@ -229,21 +279,33 @@ def add_control_point(self, control_point):
         :return: Control point id
         :rtype: int
         """
-
+        # Where is this used? Is it to provide the user with an ability to build adhoc geometry on the fly?
         self.control_points.append(control_point)
         return len(self.control_points) - 1
 
-    def clean_geometry(self, verbose=False):
+    def clean_geometry(self, tolerance: float = 1e-6):
         """Peforms a full clean on the geometry.
 
-        :param bool verbose: If set to true, information related to the geometry cleaning process
-            is printed to the terminal.
+        :param float tolerance: If any two points are of a distance apart less than 'tolerance'
+        then the two points are "buffered" to a single point. The first point is kept and the second discarded.
 
         ..  note:: Cleaning the geometry is always recommended when creating a merged section,
           which may result in overlapping or intersecting facets, or duplicate nodes.
         """
+        if self.geom.is_valid:
+            clean_at_tol = self.geom.buffer(tolerance)
+            clean_at_zero = self.geom.buffer(0)
+            if clean_at_tol.is_valid: self.geom = clean_at_tol
+            elif clean_at_zero.is_valid: self.geom = clean_at_zero
+
+        else:
+            clean_with_union = shapely.ops.unary_union(self.geom)
+            if clean_with_union.is_valid: self.geom = clean_with_union
+            else:
+                raise ValueError("Geometry is invalid after attempting shapely.ops.unary_union. Please review geometry.")
+            
```

Comment:
The logic here is pretty loose-y/goose-y because I have been unsure about how to deal with compound sections (e.g. MergedSection). Essentially, I am trying out two different methods of cleaning geometry within shapely: "zero buffer" and "union of all objects". I still need to work out how this will work (also as part of working out CrossSection and MergedSection).